core/runtime/v2/shim/util_unix.go

```diff
@@ -177,3 +197,85 @@ func CanConnect(address string) bool {
 	conn.Close()
 	return true
 }
+
+func HybridVsockDialer(addr string, port uint64, timeout time.Duration) (net.Conn, error) {
+	timeoutCh := time.After(timeout)
+	for {
+		conn, err := net.DialTimeout("unix", addr, timeout)
+		if err != nil {
+			return nil, err
+		}
+		if _, err = conn.Write([]byte(fmt.Sprintf("CONNECT %d\n", port))); err != nil {
+			conn.Close()
+			return nil, err
+		}
+		errChan := make(chan error)
+		go func() {
+			reader := bufio.NewReader(conn)
+			response, err := reader.ReadString('\n')
+			if err != nil {
+				errChan <- err
+				return
+			}
+			if strings.Contains(response, "OK") {
+				errChan <- nil
+			} else {
+				errChan <- fmt.Errorf("hybrid vsock handshake response error: %s", response)
+			}
+		}()
+		select {
+		case err = <-errChan:
+			if err != nil {
+				conn.Close()
+				// when it is EOF, maybe the server side is not ready.
+				if err == io.EOF {
+					time.Sleep(10 * time.Millisecond)
```

Comment:
10 a bit too much IMO.
Should we reduce this or at least add a warning message.

Reply:
Changed this value to `timeout/10` to do 10 retries before timeout. and warning log is added.